From 1cf6c4bdfb8994fa471e1f0e9c8ccde7cdd3614d Mon Sep 17 00:00:00 2001
From: JerryCasiano <JerryCasiano@gmail.com>
Date: Sat, 14 Sep 2019 16:30:59 -0400
Subject: [PATCH] - Fix segfault due to difference in code generated by newer
 Vala versions - Closes #96 - Fix warning related to delegates also introduced
 in newer Vala versions.

---
 src/font-manager/Application.vala | 2 +-
 src/font-manager/Compare.vala     | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/font-manager/Application.vala b/src/font-manager/Application.vala
index 16282f6..ec5b85c 100644
--- a/src/font-manager/Application.vala
+++ b/src/font-manager/Application.vala
@@ -448,7 +448,7 @@ namespace FontManager {
                 Idle.add((owned) callback);
                 return true;
             };
-            new Thread <bool> ("refresh_async", run_in_thread);
+            new Thread <bool> ("refresh_async", (owned) run_in_thread);
             yield;
             return;
         }
diff --git a/src/font-manager/Compare.vala b/src/font-manager/Compare.vala
index 4680a89..dfbadbe 100644
--- a/src/font-manager/Compare.vala
+++ b/src/font-manager/Compare.vala
@@ -81,8 +81,8 @@ namespace FontManager {
             treeview.append_column(column);
             treeview.set_model(store);
             set_default_button_relief(controls);
-            foreground_color = fg_color_button.rgba;
-            background_color = bg_color_button.rgba;
+            foreground_color = fg_color_button.get_rgba();
+            background_color = bg_color_button.get_rgba();
             bind_properties();
             connect_signals();
             base.constructed();
